File: 2024/dec04/Python/01.py
import aux04 as aux

XmasCount=0
for linh in range(0, aux.howManyLines+1):
    for coln in range(0, aux.howManyCols+1):
        XmasCount+=1 if aux.check_horizontal(aux.lines, linh, coln) else 0
        XmasCount+=1 if aux.check_vertical(aux.lines, linh, coln) else 0
        XmasCount+=1 if aux.check_diagonal(aux.lines, linh, coln) else 0
        XmasCount+=1 if aux.check_antidiagonal(aux.lines, linh, coln) else 0

print("XMAS Count (Challenge 1):", XmasCount)
#Right answer: 2557

File: 2024/dec04/C/01.c
#include "dec04.h"

int main() {
    FILE *file;
    char input[ROWS][COLUMNS] = {0};
    int r=0,c=0;
    char ch;

    file=fopen("../input.txt","r");
    if (file==NULL) return 1;
    
    //prepare 2D array
    while ((ch = fgetc(file)) != EOF && r<ROWS) {
        if (ch == '\n') {
            r++;
            c=0;
        }
        else {
            input[r][c]=ch;
            c++;
        }
    }

    int XmasCount=0;

    for (int linh=0; linh < ROWS; linh++) {
        for (int coln=0; coln < COLUMNS; coln++) {
            if (check_horizontal(input,linh,coln)) XmasCount++;
            if (check_vertical(input,linh,coln)) XmasCount++;
            if (check_diagonal(input,linh,coln)) XmasCount++;
            if (check_anti_diagonal(input,linh,coln)) XmasCount++;
        }
    }

    printf("XMAS Count (Challenge 1): %d\n", XmasCount);
    //Right answer: 2557

    return 0;
}